src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -948,12 +976,16 @@ static boolean isLegalName(String name) {
                                                  FileWrapper parent,
                                                  boolean overwrite,
                                                  UserContext userContext) {
-        setModified();
         if (! isLegalName(newFilename))
             return CompletableFuture.completedFuture(parent);
+        if (! parent.isWritable())
+            return Futures.errored(new IllegalStateException("You cannot rename something without write access to the parent!"));
```

Comment:
Drop the 'you' from start of sentence. Use 'Unable to rename..'

Reply:
Done